client/src/app/shared/form-card/form-card.component.ts

```diff
@@ -191,14 +221,35 @@ export class FormCardComponent {
     this.formCardService.closeFormCard();
   }
 
+  public isDisabled(): boolean {
+    if (this.type === 'Account') {
+      return this.accountForm.invalid;
+    } else if (this.type === 'Transaction') {
+      return this.transactionForm.invalid || !this.transactionType;
+    } else if (this.type === 'Category') {
+      return this.categoryForm.invalid || !this.categoryType;
+    }
+
+    return true;
+  }
+
   constructor(
     public formCardService: FormCardService,
     private accountService: AccountService,
     private transactionService: TransactionService,
     private reloadService: ReloadService,
     public currencyService: CurrencyService,
-    private categoryService: CategoryService,
+    public categoryService: CategoryService,
     private modalService: ModalService,
     private notificationService: NotificationService
   ) {}
+
+  ngOnInit(): void {
+    this.categoryService.getCategories().subscribe({
```

Comment:
untilDestroyed

Reply:
its updated in new commits, as you said i did not left single subscribtion without untilDestroy :d